### PR TITLE
feat: Migrate from bugsnag to sentry

### LINKED ISSUE #1318: Evaluate moving to Sentry from BugSnag
**Author the user story for this feature**

As a Flank developer, I want to understand crashes in Flank so I can fix them.

**Describe the solution**

Evaluate [Sentry](http://sentry.io/) as a BugSnag alternative.

### PR DESCRIPTION
**Pull request by adamfilipow92**
Fixes #1318 

1. Flank should display session id below ```revision```
2. Should save session id to ```session_id.txt``` in matrix path
3. Should upload ```session_id.txt``` when ```disable-results-upload``` not set
4. Unit & Integration tests should pass

## Checklist

- [X] Save and upload session id
- [X] Unit tested
- [X] Integration tests updated

### CODE REVIEW

File: buildSrc/src/main/kotlin/Versions.kt [hunk 1]
```diff
@@ -2,6 +2,9 @@ object Versions {
     // https://github.com/bugsnag/bugsnag-java/releases
     const val BUGSNAG = "3.6.2"
```
Comment (axelzuziak-gogo) [inline buildSrc/src/main/kotlin/Versions.kt:3]: Can we remove it?

**Comment by adamfilipow92** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by adamfilipow92** [comment]
recheck

File: test_runner/src/main/kotlin/ftl/util/CreashReporter.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+internal fun initCrashReporter(
+    useMock: Boolean,
+    rootPath: String = System.getProperty("user.home")
+) = when {
+    useMock -> null
+    analyticsFileExistAndIsDisabled(rootPath) -> null
+    else -> initializeCrashReportWrapper()
+}
+
+private fun analyticsFileExistAndIsDisabled(rootPath: String) =
```
Comment (pawelpasterz) [inline test_runner/src/main/kotlin/ftl/util/CreashReporter.kt:34]: Hm...this name is not fully self-explanatory (for me). The logic here is to check if a user had disabled google analytics, flank should respect those settings.
How about `isGoogleAnalyticsDisabled`? Maybe not exactly the same but something explains why we check that file

File: test_runner/src/main/kotlin/ftl/util/CreashReporter.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
+package ftl.util
```
Comment (pawelpasterz) [inline test_runner/src/main/kotlin/ftl/util/CreashReporter.kt:1]: Small typo in the file name `CreashReporter` -> `CrashReporter`

File: test_runner/src/main/kotlin/ftl/gc/GcStorage.kt [hunk 4]
```diff
@@ -101,6 +103,14 @@ object GcStorage {
         )
     }
 
+    fun IArgs.uploadSessionId() = takeUnless { disableResultsUpload }?.let {
+        upload(
+            file = Path.of(localResultDir, SESSION_ID_FILE).toString(),
```
Comment (pawelpasterz) [inline test_runner/src/main/kotlin/ftl/gc/GcStorage.kt:108]: We use `Paths` in other places in the project. Your implementation is not wrong! I am just wondering if it is possible to use `Paths` here as well

File: test_runner/src/main/kotlin/ftl/util/CreashReporter.kt (hunk 2 above)
Comment (adamfilipow92) [reply test_runner/src/main/kotlin/ftl/util/CreashReporter.kt:34]: agree

File: test_runner/src/main/kotlin/ftl/gc/GcStorage.kt (hunk 4 above)
Comment (adamfilipow92) [reply test_runner/src/main/kotlin/ftl/gc/GcStorage.kt]: Changed! Thanks! 👍

**Review by pawelpasterz — APPROVED** [review]

**Review by piotradamczyk5 — APPROVED** [review]

**Pull request merged**